docs/surveys/2019-10_3-months-of-govuk-docker.md

```diff
@@ -0,0 +1,291 @@
+
+### How could GOV.UK Docker meet more of your needs?
+
+- More data replication functionality (6 responders).
+- Not all apps are there (4 responders).
+- Easier to use (2 responders).
+- Be able to test govuk-puppet locally.
+- It would be nice to have some sample data in the databases (eg. a few
+  documents in different states).
+- Making service ports (eg, elasticsearch) exposed to the host by default
+  would be handy.
+- To setup Whitehall I needed to run the Publishing API separately or it'd
+  fail. To run Whitehall frontend I need to run Static separately or a page
+  won't render.
+
+### Are there aspects of GOV.UK Docker that cause you problems?
+
+- Issues running since Ruby upgrade (3 responders).
+- Debugging why an app isn't working locally can be difficult.
+- Setup of the databases was tricky.
+- Sometimes dependant apps fail to start up.
+- Workers aren't run by default with an app.
+- Initial app build takes time.
+- Static has problems running.
+- I often get a red failed to connect to service error message.
+- It is a little tricky to set up and usually needs a bit of effort to get to
+  work.
+- The development process could be more transparent.
+- Limited documentation for completely new starters.
+- It also seems to take up quite a lot of space for each app.
+- Troubleshooting difficult as I don't know the docker well.
+
+## Questions for people who don't use GOV.UK Docker
+
+### Why don't you use GOV.UK Docker?
+
+- Mostly run apps on local machine with `--live` flag (3 responders)
+- I haven't had the opportunity to. My VM still works.
+- I missed firebreak and don't know why we're deprecating the dev VM.
+- Every time I come to use it I can never get it running first time. Usually a
+  lot of errors and takes a lot of effort to fix them.
+- Conflicts with BYOD VPN
+
+### Have you tried using GOV.UK Docker?
+
+| Answer                                                         | Responses | Percentage |
+|----------------------------------------------------------------|-----------|------------|
+| I did, it seemed to work                                       | 3         | 33.3%      |
+| Yes, but it wasn't working properly                            | 3         | 33.3%      |
+| I tried but got stuck at an installation/initial setup problem | 2         | 22.2%      |
+| No I haven't                                                   | 1         | 11.1%      |
+
+### Are you interested in using GOV.UK Docker for your development?
+
+| Answer | Responses | Percentage |
+|--------|-----------|------------|
+| Yes    | 6         | 66.7%      |
+| No     | 3         | 33.3%      |
+
+#### Reasons for yes
+
+- Easier to develop locally and is quicker compared to the VM.
+- I helped do some of the initial work to get it going, seemed pretty good and
+  more usable than the VM.
+- If I have to run applications with more dependencies.
+- If it solves the problem that you can't use the VM and VPN at the same time.
```

Comment:
Interesting that this observation conflicts with a previous one!

Reply:
This difference is because there are two VPNs: the GDS Developer VPN (for GDS Macs) and the BYOD VPN (for GDS Linux devices, and people's personal devices).

The Developer VPN has a split-tunnel mechanism such that non-local traffic goes via the VPN and local traffic (dev VMs, Docker, etc.) doesn't. The BYOD VPN doesn't trust anything and tunnels all traffic through, which obviously means that local development doesn't work at all. (This is the reason I've had to go back to using my Mac as opposed to continuing with the GDS Linux laptops trial.)

The #network-linux community Slack channel is in talks with IT about changing the BYOD VPN behaviour to work like the Developer VPN, such that using Docker is possible - it's a known frustration.

Reply:
Thanks for explaining @issyl0 😄. As a Mac-only user, I've obviously been spoiled with just the Developer VPN.